modules/custom/express_dashboards/express_dashboard_lock/express_dashboard_lock.module

```diff
@@ -0,0 +1,241 @@
+  return $output;
+}
+
+/**
+ * Implements hook_form() and builds out lock users options on admin/dashboard/lock.
+ */
+function express_dashboard_lock_admin_form($form, &$form_state) {
+  global $user;
+  $form['instructions'] = array(
+    '#type' => 'markup',
+    '#title' => t('Lock instructions'),
+    '#markup' => t('
+      <p>This form allows you to lock users from making changes to the site. Users will still be able to log in, but 
+       certain pages and forms will be disabled for them. They will not be able to add or edit content and will see the
+       message you set below when they try to access a page or form that has been locked.</p>
+       <p>It\'s recommended to leave contact information in the message below so that users can contact you about the 
+       site.</p>
+    ')
+  );
+  $form['lock_by_role'] = array(
+    '#type' => 'checkboxes',
+    '#multiple' => TRUE,
+    '#title' => t('Lock users based on their role'),
+    '#options' => array(
+      // The keys for the array are user role ID #s.
+      '6' => t('Edit my content'),
+      '4' => t('Content editors'),
+      '7' => t('Site owners'),
+      '3' => t('Administrators'),
+    ),
+    '#default_value' => variable_get('lock_by_role', array()),
+    '#description' => t('Every user with a role selected above will be locked out.'),
+  );
+  $form['lock_allow_users'] = array(
+    '#type' => 'textfield',
+    '#multiple' => TRUE,
+    '#title' => t('Allow specific users to be exempt from their role\'s lock'),
+    '#default_value' => variable_get('lock_allow_users'),
+    '#description' => t('Separate usernames with spaces only. For example, "user1 user2 user3".'),
+  );
+  $form['lock_leave_message'] = array(
+    '#type' => 'textarea',
+    '#title' => t('Leave a message for locked out users'),
+    '#default_value' => variable_get('lock_leave_message', t('This site has been locked. For more information contact the Site Owner.')),
+  );
+  $form['submit'] = array(
+    '#type' => 'submit',
+    '#value' => t('Save configuration'),
+  );
+  return $form;
+}
+
+/**
+ * Implements hook_form_submit().
+ *
+ * You can't lock yourself out while submitting the form.
+ */
+function express_dashboard_lock_admin_form_submit($form, &$form_state) {
+  global $user;
+  // Exclude unnecessary elements.
+  form_state_values_clean($form_state);
+
+  foreach ($form_state['values'] as $key => $value) {
+    if (is_array($value) && isset($form_state['values']['array_filter'])) {
+      $value = array_keys(array_filter($value));
+    }
+    variable_set($key, $value);
+  }
+
+  $dont_lock_yourself_out_when_submitting = $form_state['values']['lock_allow_users'];
+  $dont_lock_yourself_out_when_submitting .= t(' ') . $user->name;
+
+  $allowed_users = explode(' ', variable_get('lock_allow_users'));
+
+  if (!in_array($user->name, $allowed_users)) {
+    variable_set('lock_allow_users', $dont_lock_yourself_out_when_submitting);
+  }
+
+  drupal_set_message(t('Lock has been enabled.'));
```

Comment:
Adding users by username could be a tricky task and prone to errors so I would appreciate a validation step as a user where you check and see if the strings of usernames I've entered are users and spit me back an error if I've made a mistake. Otherwise, I could think I've locked someone out and not know I made a mistake until they take down my site, lol.